mlir/test/Pass/nested-operations.mlir

```diff
@@ -0,0 +1,24 @@
+// RUN: mlir-opt -allow-unregistered-dialect %s -affine-scalrep | FileCheck %s
+
+// Affine scalar replacement pass is a func::FuncOp pass.
+// Ensure that simple_store_load function nested in gpu.module can be transformed by -affine-scalrep.
+
+gpu.module @functions {
+  // CHECK-LABEL: func @simple_store_load() {
+  func.func @simple_store_load() {
```

Comment:
Can you add a gpu.func as well to ensure that the same transformation (affine scalar replacement) can be performed as expected.  I think we should have 2 cases:
  - gpu.func with a kernel attribute
  - gpu.func without kernel attribute